### ISSUE TITLE
Using Apple Sign-On with PFLoginViewController

### ISSUE THREAD

**Issue by brabe79** [parse-community/Parse-SDK-iOS-OSX#1493]
Has anyone successfully added the apple sign on button with PFLoginViewController? My app is getting rejected because I don't have Apple Signon and I'vey built everything with PFLoginViewController. Just curious if anyone has any examples. Thanks!

**Comment by dbergman** [comment]
I am also looking for a solution for this.

**Comment by drdaz** [comment]
Unfortunately the current solution is to make your own login view controller and implement it there.

I was working on a branch that added the Sign In With Apple button to PFLoginViewController, but I won't merge it until the issues with our master branch are fixed.

**Comment by UnderratedDev** [comment]
I can take a look at this. In reference to this: [Parse Server Pull Request](https://github.com/parse-community/parse-server/pull/6523). I made the backend changes for Apple Sign In for Parse Server & have implemented my own Apple Sign In with Parse. I can do some cleanup & perhaps release a utils file, & take a look at the PFLoginController also while I'm doing that.

**Comment by noobs2ninjas** [comment]
@UnderratedDev That would be amazing! We actually are trying to get bolts and dependencies updated to work with iOS 13/Catalyst submissions and had to put off our AppleSignIn api off for the mean time. However, thats definitely our biggest request right now. So, any help would be amazing.

**Comment by drdaz** [comment]
> I can take a look at this. In reference to this: [Parse Server Pull Request](https://github.com/parse-community/parse-server/pull/6523). I made the backend changes for Apple Sign In for Parse Server & have implemented my own Apple Sign In with Parse. I can do some cleanup & perhaps release a utils file, & take a look at the PFLoginController also while I'm doing that.

I've done most of the implementation for Parse's architecture in the branch here called Apple_Sign_In. Take a look if you want.

Since starting it there were significant changes to the project files in master. So there needs to be either a serious merge, or a manual move of the new code content into a more recent branch. The latter option is probably less painful.

**Comment by UnderratedDev** [comment]
@drdaz Thanks man, I appreciate it, I'll definitely check that out. I did a base implementation of the provider a while ago but it needs a lot of work, so I will mostly likely use yours as a start point or for reference! :)

**Comment by noobs2ninjas** [comment]
@UnderratedDev We've been talking about a big ParseUI update for 1.18.1  for a while. We're wanting to get them all updated so developers can use the newer features that comes with iPad OS and Catalyst. Or at the minimum just get them usable in catalyst in general Definitely a discussion for later, just something to keep in mind as you look through everything. Ideally though, we'd love to go through making sure that the login view and authentication works great in Slide Over and works seamlessly with split/multi-window setups. Not that you have to do all that, just thought it might be something to think about as you go through everything.

**Comment by drdaz** [comment]
I have every intention of getting my Apple Sign In code finished and merged fwiw (I spent a lot of time on it). It just made no sense to work on it while master was broken.

**Comment by noobs2ninjas** [comment]
Thats true @drdaz. Do you think the code is large enough to warrant its own submodule like Twitter and FacebookUtils?

**Comment by drdaz** [comment]
Apple Sign In? I don't think so. The FB and Twitter Utils projects act as facades. The Apple Sign In implementation is much simpler than the FB and Twitter logins since the logic is baked into the OS. I don't feel like it's necessary to wrap it to the same extent.

**Comment by isaacAssaban** [comment]
> Unfortunately the current solution is to make your own login view controller and implement it there.

Hey @drdaz would you mind sharing your apple sign in implementation in custom vc with parse ?. I have got the apple sign in flow working but I dont really know how i should create my own user with data retrieved from apple credentials. 
Calling  PFUser.logInWithAuthType(inBackground: "apple", authData: ["token":tokenString, "id": user]) provkes an internal server error coming from parse. 
Optional("{\"code\":1,\"message\":\"Internal server error.\"}")

Would be greatly appreciated, thx

**Comment by brianyyz** [comment]
@drdaz would also like to see your implementation if you can share.

**Comment by drdaz** [comment]
Alright. Full disclaimer; I don't know the exact state of this. It's been months since I looked at it. But here https://github.com/parse-community/Parse-SDK-iOS-OSX/tree/Apple_Sign_In

If you're just trying to understand and implement Apple Sign In (not using ParseUI), this might not be the ideal base to look at; there's going to be some misdirection going on due to Parse's architecture.

**Comment by drdaz** [comment]
Oh I just realised that's not what you're asking for... ahh. I can't share my own (non ParseUI) implementation right now. But it was based on Back4App's sample code: https://www.back4app.com/docs/ios/sign-in-with-apple-ios

**Comment by brianyyz** [comment]
Thanks for the pointer!

**Comment by isaacAssaban** [comment]
thanks for the quick reply @drdaz 
Yeah I have been using the back4app sample code + configuration from the tutorial but whatever I do, I end up receiving the same message from Parse. Optional("{"code":1,"message":"Internal server error."}")

I just downloaded the sample again, created an appId and added it in parse Apple Settings, created a service id, and a key. 
I have also enabled the apple sign in my service id, and configured the web auth configuration by adding "APP_NAME".back4app.com as the domain URL and https://"APP_NAME".back4app.com/redirect as the redirect URL.

Finally i have added the appId (bundle id) in Parse Apple Settings, and modified the appdelegate with my appId and clientKey
However once i run the sample project and sign with apple, parse returns an internal server error upon loginWithAuth call. 

I noticed during the web auth configuration in the service ID there was no apple-developer-domain-association.txt file to download anymore. Might it be related to this, am I missing something ? Any help would be greatly appreciated.
